jmclient/jmclient/cryptoengine.py

```diff
@@ -61,6 +63,21 @@ def is_extended_public_key(key_str):
         BIP32_PUB_PREFIX, BIP49_PUB_PREFIX, BIP84_PUB_PREFIX, TESTNET_PUB_PREFIX]])
 
 
+def convert_xpub_if_needed(xpub):
+    if not xpub.startswith(BIP32_PUB_PREFIX) or not qt_display_ypub_zpub():
+        return xpub
+
+    # Convert xpub to ypub or zpub for display
+    if is_segwit_mode():
+        if is_native_segwit_mode():
```

Comment:
This is wrong, should look at the wallet type not `joinmarket.cfg` settings. Noticed by opening native segwit JM wallet with `native = false` in `joinmarket.cfg`.

Reply:
Fixed.